### PR TITLE
CLOUDP-127622: Add e2e test for "atlas clusters upgrade" command

### PR DESCRIPTION
**Pull request by Razeer123**
## Proposed changes

Adds e2e test for "atlas clusters upgrade" command.

_Jira ticket:_ [CLOUDP-127622](https://jira.mongodb.org/browse/CLOUDP-127622)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by fmenezes — APPROVED** [review]
Excellent work!

File: test/e2e/atlas/clusters_upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas/mongodbatlas"
+	exec "golang.org/x/sys/execabs"
+)
+
+func TestSharedClusterFlags(t *testing.T) {
```
Comment (gssbzn) [inline test/e2e/atlas/clusters_upgrade_test.go:31]:
```suggestion
func TestSharedClusterUpgrade(t *testing.T) {
```

File: test/e2e/atlas/clusters_upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	region, err := g.newAvailableRegion(tierM2, e2eClusterProvider)
+	req.NoError(err)
+
+	t.Run("Create", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			"create",
+			clusterName,
+			"--region", region,
+			"--members=3",
```
Comment (gssbzn) [inline test/e2e/atlas/clusters_upgrade_test.go:51]: should be safe to delete for a shared tier
```suggestion
```

File: test/e2e/atlas/clusters_upgrade_test.go (hunk 1 above)
Comment (Razeer123) [reply test/e2e/atlas/clusters_upgrade_test.go:31]: Ohh, I didn't see that, thanks!

**Pull request merged**